### PR TITLE
support virtio network device on windows 

### PR DESCRIPTION
**Pull request by thehajime**
This is to introduce networking support with virtio device on windows platform.
The implementation relies on [tap-windows6](https://github.com/OpenVPN/tap-windows6) framework from openvpn over msys2 environment.  Thus patchset contains some of fixes for msys2 build as well as CI platform over github actions.

### CODE REVIEW

File: tools/lkl/tests/disk.sh [hunk 1]
```diff
@@ -9,6 +9,8 @@ function prepfs()
     set -e
 
     file=`mktemp`
```
Comment (tavip) [inline tools/lkl/tests/disk.sh:11]: We could use something like file=`mktemp disk-XXXX` to create the image file in the current directory

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -7,16 +7,107 @@ on:
...
+          BRANCH="${GITHUB_REF/#refs\/heads\//}"
+
+          git init .
+          git remote add origin $REPO
+          git config core.sparseCheckout true
+          echo -e '/*\n!*aux.*' >> .git/info/sparse-checkout
+          git -c protocol.version=2 fetch --no-tags --prune --progress --depth=10 origin +${GITHUB_SHA}:refs/remotes/origin/${BRANCH}
+          git checkout --progress --force -B $BRANCH refs/remotes/origin/$BRANCH
+
+      - uses: actions/cache@v2
```
Comment (tavip) [inline .github/workflows/ci.yml:81]: Nice!

File: tools/lkl/scripts/checkpatch.sh [hunk 3]
```diff
@@ -17,8 +17,8 @@ if [ -z "$upstream" ]; then
 fi
 
 if [ -z "$lkl" ]; then
-    echo "can't find lkl remote, quiting"
-    exit 1
+    git remote add lkl-upstream https://github.com/lkl/linux || true
+    lkl=`git remote -v | grep lkl | cut -f1 | head -n1`
```
Comment (tavip) [inline tools/lkl/scripts/checkpatch.sh:21]: Should we grep with $LKL here to be as specific as possible?

File: tools/lkl/scripts/checkpatch.sh [hunk 4]
```diff
@@ -17,8 +17,8 @@ if [ -z "$upstream" ]; then
...
 fi
 
 git fetch $lkl
```
Comment (tavip) [inline tools/lkl/scripts/checkpatch.sh:24]: Also I think we need to check again for [ -z "$lkl" ] as before my patch, sorry about the mess.

**Review by tavip — APPROVED** [review]
LGTM, some minor comments inline. Great new feature!

File: tools/lkl/tests/disk.sh (hunk 1 above)
Comment (thehajime) [reply tools/lkl/tests/disk.sh]: thanks, modified as you suggested.

File: tools/lkl/scripts/checkpatch.sh (hunk 4 above)
Comment (thehajime) [reply tools/lkl/scripts/checkpatch.sh]: fixed it as well.

File: tools/lkl/scripts/checkpatch.sh (hunk 3 above)
Comment (thehajime) [reply tools/lkl/scripts/checkpatch.sh]: fixed this.

**Pull request merged**